### PR TITLE
Adding campaign export applet and minor adjustments

### PR DESCRIPTION
**Pull request by danielmgit**
Minor adjustments to current setup.

### CODE REVIEW

File: applets/008-export-campaign/applet.py [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+               "mkdir -p {}".format(dic["file_path"]))
+    # save campaign file
+    campaign_file = open(os.path.join(dic["file_path"], dic["file_name"]), "w")
+    campaign_file.write(campaign_string)
+    campaign_file.close()
+    return True
+
+
+def create_file_path(x, file_descriptor, absolute, dic):
+    file_path = str()
```
Comment (hgn) [inline applets/008-export-campaign/applet.py:46]: not required, right? Both lines

File: applets/008-export-campaign/applet.py [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+        except IndexError:
+            # this means the top directory is used
+            # will throw a permission denied anyways
+            file_path = "/"
+
+    if not (file_path or file_name):
+        x.p.err("error: file path and/or name is/are empty\n")
+        return False
+
+    dic["file_path"] = file_path
```
Comment (hgn) [inline applets/008-export-campaign/applet.py:75]: not "dic". Better name it what it contains, not like the datastructure is. And the key is better separated by dash, not underscore (this is not windows here)

File: applets/008-export-campaign/applet.py [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+        file_name = file_path_split[len(file_path_split) - 1]
+        file_path_split.pop(len(file_path_split) - 1)
+        file_path = "/".join(file_path_split)
+    # absolute file path logic
+    elif absolute:
+        file_path_split = file_descriptor.split("/")
+        file_name = file_path_split[len(file_path_split) - 1]
+        try:
+            file_path_split.pop(len(file_path_split) - 1)
+            file_path = "/".join(file_path_split)
```
Comment (hgn) [inline applets/008-export-campaign/applet.py:65]: os.path.join() ?

File: applets/008-export-campaign/applet.py [hunk 4]
```diff
@@ -0,0 +1,118 @@
...
+    # path and file name shuffling magic
+    if not absolute:
+        current_file_dir = os.path.dirname(__file__)
+        # nas file path + dir and filename to dst file
+        file_path_full = os.path.join(current_file_dir + "/../../",
+                                      file_descriptor)
+        file_path_split = file_path_full.split("/")
+        file_name = file_path_split[len(file_path_split) - 1]
+        file_path_split.pop(len(file_path_split) - 1)
+        file_path = "/".join(file_path_split)
```
Comment (hgn) [inline applets/008-export-campaign/applet.py:58]: os.path.join() ?

File: applets/008-export-campaign/applet.py [hunk 5]
```diff
@@ -0,0 +1,118 @@
...
+    file_path = str()
+    file_name = str()
+    # note: the current file path is two levels below nas
+    # path and file name shuffling magic
+    if not absolute:
+        current_file_dir = os.path.dirname(__file__)
+        # nas file path + dir and filename to dst file
+        file_path_full = os.path.join(current_file_dir + "/../../",
+                                      file_descriptor)
+        file_path_split = file_path_full.split("/")
```
Comment (hgn) [inline applets/008-export-campaign/applet.py:55]: "/" -> os.path.sep? I mean it is linux anyway and can be done in your way, but I am not soooo[TM] happy

File: applets/008-export-campaign/applet.py (hunk 1 above)
Comment (danielmgit) [inline applets/008-export-campaign/applet.py:46]: I'd say, they are.

71    if not (file_path or file_name):
72        x.p.err("error: file path and/or name is/are empty\n")
73        return False

File: applets/008-export-campaign/applet.py (hunk 2 above)
Comment (danielmgit) [inline applets/008-export-campaign/applet.py:75]: nice hint

File: applets/008-export-campaign/applet.py (hunk 3 above)
Comment (danielmgit) [inline applets/008-export-campaign/applet.py:65]: "TypeError: join() argument must be str or bytes, not 'list'". I'll have a look at it.

File: applets/008-export-campaign/applet.py (hunk 4 above)
Comment (danielmgit) [inline applets/008-export-campaign/applet.py:58]: as above

File: applets/008-export-campaign/applet.py (hunk 5 above)
Comment (danielmgit) [inline applets/008-export-campaign/applet.py:55]: I also like doing it the "correct" way. The question is if the trade-off in additional work is worth it.I'll have a look at it. Thanks for the hint anyway.

Comment (danielmgit) [inline applets/008-export-campaign/applet.py:55]: Update: I just saw that I did the path logic poorly, and by poorly I mean like extremely shitty poorly, anyway. At least it was working.

**Pull request merged**